### PR TITLE
Jenkins Application to Run on Threefold Grid

### PR DESCRIPTION
**Pull request by amr-yassir**
### Description

Added Jenkins Application to the Threefold Dashboard, allowing users to deploy Jenkins instances on the Grid.
Enabled passing Jenkins configuration parameters through environment variables.

Screenshot 2024-08-25 183514
Screenshot 2024-08-25 183338

### Related Issues

- codescalersinternships/home/issues/245

### Documentation PR

- https://github.com/threefoldtech/info_grid/pull/629

### Checklist

- [ ] Tests included
- [x] Build pass
- [x] Documentation
- [x] Code format and docstrings
- [x] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

File: packages/playground/src/weblets/tf_deployment_list.vue
Comment (zaelgohary) [inline packages/playground/src/weblets/tf_deployment_list.vue:460]: Action buttons should be added in the template.

File: packages/playground/src/weblets/jenkins.vue
Comment (zaelgohary) [inline packages/playground/src/weblets/jenkins.vue]:
```suggestion
const name = ref(generateName({ prefix: "jk" }));
```

File: packages/playground/src/weblets/jenkins.vue
Comment (zaelgohary) [inline packages/playground/src/weblets/jenkins.vue]: Use generatePassword.

File: packages/playground/src/weblets/jenkins.vue
Comment (zaelgohary) [inline packages/playground/src/weblets/jenkins.vue]:
```suggestion
const planetary = ref(true);
```

File: packages/playground/src/constants/deployment_list.ts
Comment (zaelgohary) [inline packages/playground/src/constants/deployment_list.ts:176]: Include all env vars.

File: packages/playground/src/weblets/jenkins.vue
Comment (zaelgohary) [inline packages/playground/src/weblets/jenkins.vue]:
```suggestion
    layout.value.setStatus("deploy", "Rolling back due to failure to deploy the gateway...");
```

**Review by zaelgohary — CHANGES_REQUESTED** [review]

**Comment by zaelgohary** [comment]
I can't ssh to the machine.

image

image

**Comment by zaelgohary** [comment]
Image specs should match.

image

image

**Comment by zaelgohary** [comment]
Add planetary here too.

image

File: packages/playground/public/info/jenkins.md
Comment (zaelgohary) [inline packages/playground/public/info/jenkins.md]: The link should be the one produced from the manual. It will be ''https://www.manual.grid.tf/documentation/dashboard/solutions/jenkins.html"

**Comment by amr-yassir** [comment]
I noticed that all of the implemented applications use `Mycelium and IPv4 as networks, so would you like to initiate a new constant to display the Planetary network instead of the IPv4 specifically in this application?`

**Comment by amr-yassir** [comment]
> Add planetary here too.
> 
> image



> Add planetary here too.
> 
> image

I noticed that all of the implemented applications use `Mycelium` and `IPv4` as networks, so would you like to initiate a new constant to display the `Planetary` network instead of the `IPv4` specifically in this application?

**Comment by amr-yassir** [comment]
> Add planetary here too.
> 
> image

I noticed that all of the implemented applications use `Mycelium` and `IPv4` as networks, so would you like to initiate a new constant to display the `Planetary` network instead of the `IPv4` specifically in this application?

For now, I added IPv4

**Comment by zaelgohary** [comment]
> > Add planetary here too.
> 
> > 
> 
> > image
> 
> 
> 
> I noticed that all of the implemented applications use `Mycelium` and `IPv4` as networks, so would you like to initiate a new constant to display the `Planetary` network instead of the `IPv4` specifically in this application?
> 
> 
> 
> For now, I added IPv4

Display ipv4 but make planetary's default value true.

**Comment by zaelgohary** [comment]
@maxux Can you promote this flist? https://hub.grid.tf/amryassir.3bot/amryassir-jenkinsv3-latest.flist

**Review by zaelgohary — COMMENTED** [review]
All good! will approve once flist gets promoted & doc PR gets merged. Good job! @amr-yassir

**Comment by maxux** [comment]
Which target name ? `tf-official-apps/jenkinsv3-latest.flist` ?

**Comment by zaelgohary** [comment]
@maxux `tf-official-apps/jenkins-latest.flist`

File: packages/playground/src/weblets/jenkins.vue
Comment (0oM4R) [inline packages/playground/src/weblets/jenkins.vue]:
```suggestion
         (name: string) => validators.isAlpha('Name must start with an alphabetical character.')(name[0]),
```


as we agreed on [#3287 ](https://github.com/threefoldtech/tfgrid-sdk-ts/pull/3287#discussion_r1730291955)

**Review by 0oM4R — CHANGES_REQUESTED** [review]
great solution! thank you for the great job
I think we have to follow up the Wireguard requirements in #3287

File: packages/playground/src/weblets/jenkins.vue
Comment (0oM4R) [reply packages/playground/src/weblets/jenkins.vue]: please do it also on https://github.com/threefoldtech/tfgrid-sdk-ts/pull/3338/files#diff-fc3e26378b57a55a0c7e60ae1fed09033c71b6acc8f96d0a85a85e5e00b10d4cR36 as well

**Comment by maxux** [comment]
https://hub.grid.tf/tf-official-apps/jenkins-latest.flist.md
Sorry for the delay

File: packages/playground/src/weblets/jenkins.vue
Comment (0oM4R) [reply packages/playground/src/weblets/jenkins.vue]: > please do it also on https://github.com/threefoldtech/tfgrid-sdk-ts/pull/3338/files#diff-fc3e26378b57a55a0c7e60ae1fed09033c71b6acc8f96d0a85a85e5e00b10d4cR36 as well

@amr-yassir

**Review by zaelgohary — APPROVED** [review]

**Review by 0oM4R — APPROVED** [review]

**Pull request merged**
